Feature vs3 i2 c rw command master slave esp32

RoxanneMango opened the pull request:
- Read functionality has been implemented in the I2cEsp class for both master and slave mode
- Write functionality has been implemented in the I2cEsp class for both master and slave mode
- printBuffer() function has been implemented to print out buffer contents
- Getters for buffer pointers have been written
- Getter for bufferLength has been written
- 3 Demo tasks have been written in I2C_ESP32/main.cpp:
1. masterAndSlave -- can be used on one ESP-WROOM-32 to test both master and slave mode
2. master -- can be used in conjunction with the slave task on two separate ESP-WROOM-32s
3. slave -- can be used in conjunction with the master task on two separate ESP-WROOM-32s
- Important variables have been put in the I2cEsp class as private member variables
- The I2C driver on which the code is built only accepts raw pointers for the buffers -- that is why, in the constructor of the I2cEsp class, the buffers are initialized they way they are (with the new keyword). They are safely deleted again in the constructor.
- Master code is separated from slave code by the isMaster variable which determines whether an I2cEsp object is operating in slave mode or master mode. This is a parameter in the constructor, which is false by default.
- Functions and variables originating from the I2C driver on which the code is built is not formatted according to the wiki code guidelines -- for the sake of saving time this is ignored.

NielsdeWaal committed:
Configured pins for SPI-slave module

NielsdeWaal committed:
Fixed the 'sorry, unimplemented' error

NielsdeWaal committed:
Fixed SPI gpio pins. SPI object is now correctly created

NielsdeWaal committed:
Setup base for SPI transfer, doens't fully function yet.

NielsdeWaal committed:
SPI sort of working. Send buffer needs be all zero and only the first 2 bytes are received correctly

NielsdeWaal committed:
Made an output for the received data, plus a blinking led upon correct first and last byte

NielsdeWaal committed:
Basic implementation of command objects

NielsdeWaal committed:
Fixed declaration bug

NielsdeWaal committed:
Basic command implementation. Needs a lot more documentation.

NielsdeWaal committed:
Documented all the command objects.

NielsdeWaal committed:
Basic tests for the command object

NielsdeWaal committed:
Started work on the I2C driver. Does not contain functional code as of yet.

NielsdeWaal committed:
Added first work on the I2C driver. Not compiling yet.

RoxanneMango committed:
- Removed stray files which were accidentally copied from feature-VS2ReadRegister branch(?)
- Added I2C_test_main.c file
	- Slave-master relationship on the same ESP32
	- Slave-master read and write to each other
- Updated .gitignore to include build files from the ESP folder

RoxanneMango committed:
Removed sdkconfig; .gitignore did not filter it out

RoxanneMango committed:
Updated README.md to include source

RoxanneMango committed:
- Changed i2c_example_main.c to main.cpp;
- Modified #defines in main.cpp for C --> C++ type compatibility;
- Modified slave & master ports so they are of type i2c_port_t instead of int;
- Added " extern "C" " to app_main() so code is C --> C++ compatible ( and compiles );

RoxanneMango committed:
Updating README.md to remove sensor information; it's not used.

RoxanneMango committed:
- Removed #defined --> changed them into constexpressions
- Added appropriate typing to constexpressions
- Swapped all printf()s for std::couts
- Swapped malloc() statements for new
- There were no free() statements, added deletes at end of variable's life
- Cleaned output to show less so everything fits on one screen
- Removed certain delays to make prints less choppy
- Added ///< style commenting to constexpressions to know what is what exactly
- Removed unnecesary comments

RoxanneMango committed:
- Removed gaps inbetween if/while/etc statements and their round braces '('
- Placed curly braces conform wiki code standards
- Removed unnecesary commentary
- Added comments to all functions
- Renamed all functions so they are camelCased now

RoxanneMango committed:
- Removed conflicting code in src map...
> Build tests kept failing due to errors caused by SPI code which is now removed
> SPI will no longer be used anyway, instead I2C will be used

RoxanneMango committed:
- Removed old test_main.cpp
- Up to date tests will be added when branch is merged into development

RoxanneMango committed:
- Removed all test files for debugging travis
- test.sh passes all tests locally, but travis keeps failing
- " g++: error: unrecognized command line option ‘-std=c++14’ "

RoxanneMango committed:
- Updated .travis.yml file

RoxanneMango committed:
- Added more comments for constexpressions

RoxanneMango committed:
- Made a separate master implementation folder for the master task
- Made a separate slave implementation folder for the slave task
- Made a masterAndSlave folder for the twoModesOneESP task
- Cleaned up code by removing mutex code which didn't do anything
- Built in error handling so the slave will not try to write to the buffer when it receives no input
- Renamed tasks for slave to 'slave' and master to 'master'
- Built in error handling when transmission buffer of slave is full, pointer will reset to start of allocated space
- Wrote slaveReadBuffer( uint8_t * data ) to wrap the library's variant and make the function call shorter
- Wrote slaveWriteBuffer( uint8_t * data ) to wrap the library's variant and make the function call shorter

RoxanneMango committed:
- Removed unused mutex code from masterAndSlave/main.cpp

RoxanneMango committed:
Merge branch 'feature-VS3-ESP32_I2C_Class' of github.com:R2D2-2018/DATA_COMMUNICATIONS into feature-VS3-I2C_RW_Command_Master-Slave_ESP32

RoxanneMango committed:
- Removed left over merge conflict messages

RoxanneMango committed:
- Clang formatted memory_map.hpp

RoxanneMango committed:
- Modified .fetch-build-environment.sh

RoxanneMango committed:
- Added I2cEsp member variables
- Added pin assignment in constructor
- Added read() function
- Added write() function
- Added modifyBufferData() function
- Added print() function
- Added getters and setters for buffers and bufferLengths
- Added masterAndSlave demo task
- Added master demo task
- Added slave demo task
> Code is not functioning properly due to buffer errors. Master will not read slave correctly
> Next commit will be complete overhaul --  starting from scratch again...

RoxanneMango committed:
- "If it looks stupid but it works it's not stupid"
- Placed functions from masterAndSlave directory directly in I2cEsp class
- WIP

RoxanneMango committed:
- Placed masterInit() and slaveInit() code in constructor of I2cEsp
- Depending whether isMaster is true || false different code will be called

RoxanneMango committed:
- Changed slave and master read functions both to overloaded read()

RoxanneMango committed:
- master READ && slave READ is working now
- master WRITE && slave WRITE is working now
- Buffers are now member variables of I2cEsp class
- portNum for master and slave are now member variables of I2cEsp class

RoxanneMango committed:
- Placed all global constexpressions inside I2cEsp class or deleted them
- Placed master and slave initialisation in constructor of I2cEsp class
- Placed master and slave read functionality in read() function in I2cEsp class
- Placed master and slave write functionality in write() function in I2cEsp class
- Wrote masterAndSlave demo task which works on one single ESP32
- Wrote master and slave demo tasks which work on two separate ESP32s
- Cleaned a lot of code
- Deleted masterAndSlave, slave, and master demo folders -- they now reside in main/I2C_ESP32/main.cpp as tasks
- Separated master and slave code -- when using I2cEsp object in master mode, slave code will not be touched, vica versa...
- Added SCL and SDA pin paramaters for I2cEsp constructor
- Added portNum parameter for I2cEsp constructor -- portNum 0 is default

RoxanneMango committed:
Merge branch 'development' of github.com:R2D2-2018/DATA_COMMUNICATIONS into feature-VS3-I2C_RW_Command_Master-Slave_ESP32

RoxanneMango committed:
- Corrected identation in I2C_ESP32.cpp buffer getter functions
- Corrected clang-format for memory_map.hpp again...

j00st commented on ESP/I2C_ESP32/main/main.cpp:
```diff
@@ -1,22 +1,199 @@
+        } else {
+			std::cout << esp_err_to_name(ret) << ": Master read slave error, IO not connected...\n";
+        }
+        ///< ---------------------------------------------------
```
Use whitespace to distinct code

j00st commented on ESP/I2C_ESP32/main/main.cpp:
```diff
@@ -1,22 +1,199 @@
+void app_main() {
+    xTaskCreate(masterAndSlave, "masterAndSlave", 1024 * 2, (void* ) 1, 10, NULL);
+    //xTaskCreate(master, "master", 1024 * 2, (void* ) 1, 10, NULL);
+    //xTaskCreate(slave, "slave", 1024 * 2, (void* ) 1, 10, NULL);
```
Remove these

RoxanneMango committed:
- Removed ///< ------ comments
- Added comments explaining what to do in app_main of ESP/I2C_ESP32/main.cpp to activate different demo code
